kernel/object-store/src/main/java/io/mosip/commons/khazana/impl/S3Adapter.java

```diff
@@ -1,580 +1,590 @@
+import java.util.Map.Entry;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+import io.mosip.commons.khazana.config.LoggerConfiguration;
+import io.mosip.commons.khazana.dto.ObjectDto;
+import io.mosip.commons.khazana.exception.ObjectStoreAdapterException;
+import io.mosip.commons.khazana.spi.ObjectStoreAdapter;
+import io.mosip.commons.khazana.util.ObjectStoreUtil;
+import io.mosip.kernel.core.exception.ExceptionUtils;
+import io.mosip.kernel.core.logger.spi.Logger;
+
+@Service
+@Qualifier("S3Adapter")
+public class S3Adapter implements ObjectStoreAdapter {
+
+    private final Logger LOGGER = LoggerConfiguration.logConfig(S3Adapter.class);
+
+    @Value("${object.store.s3.accesskey:accesskey:accesskey}")
+    private String accessKey;
+    @Value("${object.store.s3.secretkey:secretkey:secretkey}")
+    private String secretKey;
+    @Value("${object.store.s3.url:null}")
+    private String url;
+
+    @Value("${object.store.s3.region:null}")
+    private String region;
+
+    @Value("${object.store.s3.readlimit:10000000}")
+    private int readlimit;
+
+    @Value("${object.store.connection.max.retry:20}")
+    private int maxRetry;
+
+    @Value("${object.store.max.connection:200}")
+    private int maxConnection;
+
+    @Value("${object.store.s3.use.account.as.bucketname:false}")
+    private boolean useAccountAsBucketname;
+    
+	@Value("${object.store.s3.bucket-name-prefix:}")
+	private String bucketNamePrefix;
+
+    private int retry = 0;
+
+    private List<String> existingBuckets = new ArrayList<>();
+
+    private AmazonS3 connection = null;
+
+    private static final String SEPARATOR = "/";
+
+
+    @Override
+    public InputStream getObject(String account, String container, String source, String process, String objectName) {
+    	 String finalObjectName=null;
+    	 String bucketName=null;
+    	if(useAccountAsBucketname) {
+    		 finalObjectName = ObjectStoreUtil.getName(container,source, process, objectName);
+    		 bucketName=account;
+    	}else {
+    		 finalObjectName = ObjectStoreUtil.getName(source, process, objectName);
+    		 bucketName=container;
+    	}
+		bucketName = addBucketPrefix(bucketName);
+		bucketName = bucketName.toLowerCase();

```

Comment:
can you please add a comment here on why we are doing this? eg: S3 mandates to have bucketname in lowercase.

Reply:
fixed